main/src/ca/uwaterloo/flix/tools/pkg/FlixPackageManager.scala

```diff
@@ -16,25 +16,60 @@
 package ca.uwaterloo.flix.tools.pkg
 
 import ca.uwaterloo.flix.api.Bootstrap
-import ca.uwaterloo.flix.tools.pkg.Dependency.FlixDependency
+import ca.uwaterloo.flix.tools.pkg.Dependency.{FlixDependency, JarDependency, MavenDependency}
 import ca.uwaterloo.flix.tools.pkg.github.GitHub
 import ca.uwaterloo.flix.util.{Formatter, Result}
 import ca.uwaterloo.flix.util.Result.{Err, Ok, traverse}
+import ca.uwaterloo.flix.util.collection.ListMap
 
 import java.io.{IOException, PrintStream}
 import java.nio.file.{Files, Path, StandardCopyOption}
+import scala.collection.mutable
 import scala.util.Using
 
 object FlixPackageManager {
 
+  case class Resolution(origin: Manifest,
+                        manifests: List[Manifest],
+                        immediateDependents: Map[Manifest, List[Manifest]],
```

Comment:
You are using Manifest here as a key, but what is its equality? Isnt a manifest the whole file?

I would have thought dependencies are between packages?